### ISSUE TITLE
Updating the Heatmap with new labels causes an error log

### ISSUE THREAD

**Issue by andresumanzor** [apexcharts/react-apexcharts#20]
Updating the Heatmap with new labels causes an error log to be displayed as the labels are different from the ones the component used before.

Explanation:

I did create a component which does receive the renderData as a prop and then that data is passed over to the <Chart/> component but, when changing the data sent to the Heatmap, this "TypeError: Cannot read property 'color' of undefined" is displayed on the console.

This has happened to me in the past with other libraries such as "react-chartjs" but, has been solved by passing a "redraw" prop into their main Chart's component. However, I'm not able to find a workaround for this using your charts.

Can you please help me and in case there is an existing workaround already, can you please do let me know what that workaround is?

Thank you!

**Comment by junedchhipa** [comment]
Hi @andresumanzor 
Does this bug arises without React, or it happens just for React version of ApexCharts?
If you haven't tested without React, I request you do test it there first as it will be easier for me to reach the issue.
I tried in native JavaScript by calling `updateOptions()` method on Heatmap and changing the labels property, but I was not able to produce the issue.

Thanks

**Comment by andresumanzor** [comment]
Hi there @junedchhipa,

I do apologize for my late reply but, I did find out a workaround for it to work.

I have only deep tested it over the React Version and did find out that your line `var w = e.globals.previousPaths[o][u].color;` is the one causing this issue as I pass my custom component which does renders your `<Chart>` component with new arrays of series (over the `series` prop) and categories (which are located under the `options` prop) which causes the chart to break as old/new labels aren't able to be found on the array the npm does iterate through for creating the fading in/out animation.

I did fix it by changing that line to: `var w = e.globals.previousPaths[o][u] ? e.globals.previousPaths[o][u].color : '#ffffff';` determining my default color to be white in case that coordinate doesn't exist on your bidimensional array.

Thank you!
